### DISCUSSION TITLE
Implementing transpiler for python code output

### DISCUSSION THREAD

**Discussion by mnoomnoo** [mtconnect/MtconnectTranspiler#20] [General]
I was looking at Implementing `ITranspilerSink` to produce python code.

I looked at `MtconnectTranspiler.Sinks.CSharp` as an example and had a few questions:

- Whats the main difference between `MtconnectTranspiler.Sinks.CSharp` and  `MtconnectTranspiler.Sinks.CSharp.Example` ?
- Do I need to create a project similar to `MtconnectTranspiler.Sinks.CSharp` but for python in order to produce a foundation for my implementation?
    - The classes under Contracts/Interfaces like the IClass.cs or the *.cs files under Models ;  would I need to impl those for python? IClass.py, or Models/Classes/DataItemTypes/EventClass.py , etc ?
- Should my focus be on writing python templates within `MtconnectTranspiler.Sinks.CSharp.Example` ? Then running the exe with those new templates?

Thanks.

**Comment by tbm0115** [comment]
@mnoomnoo The transpiler architecture has changed as we transition from v2.4 to v2.5. In the latest release, the `MtconnectTranspiler.Sinks.CSharp` project is now intended to be a full representation of the MTConnect standard in C#. So, if you're more familiar with the C# language, you can traverse the standard using the `MtconnectTranspiler.Sinks.CSharp` library (available on NuGet).

If you're not [as] familiar with C#, then the `MtconnectTranspiler.Sinks.CSharp.Example` serves as an example of how to use the `MtconnectTranspiler` library to represent the MTConnect standard in another language (like Python). The MTConnect Institute aims to offer a full representation of the standard in Python and C++, similar to what we've done with `MtconnectTranspiler.Sinks.CSharp`.

## My recommendation
For your use case, I would recommend referencing the `MtconnectCore` ([view](https://github.com/TrueAnalyticsSolutions/MtconnectCore/tree/tbm0115/hard-coded-model/MtconnectTranspiler.Sinks.MtconnectCore)) project's latest transpiler implementation to generate the representations for MTConnect Types/SubTypes in Python.


**Do you have an example of the `.py` template you're looking to generate?**

**Reply by mnoomnoo** [reply]
I don't have an example template at this moment. 

Thanks for the reference to the MtconnectCore's transpiler sink example. 

I'll check it out and see if I can get a python transpiler working. 

Thanks
